packages/core/src/app/checkout/Checkout.test.tsx

```diff
@@ -255,6 +258,20 @@ describe('Checkout', () => {
             expect(screen.getByText(/shipping method/i)).toBeInTheDocument();
         });
 
+        it('renders custom shipping method and locks shipping component', async () => {
+            checkout.use('CartWithCustomShippingAndBilling');
+
+            render(<CheckoutTest {...defaultProps} />);
+
+            await checkout.waitForPaymentStep();
+
+            expect(screen.getByText('Manual Order Custom Shipping Method')).toBeInTheDocument();
+            expect(screen.getAllByRole('button', { name: 'Edit' })).toHaveLength(2);
```

Comment:
We are asserting these two `Edit` buttons here.

<img width="1271" alt="image" src="https://github.com/bigcommerce/checkout-js/assets/88361607/ba26b4b5-2477-4fc0-b311-461d7a6fbf56">